qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -0,0 +1,1042 @@
+        """Return experiment job IDs.
+
+        Returns: IDs of jobs submitted for this experiment.
+        """
+        return self._jobs.keys()
+
+    @property
+    def backend(self) -> Optional[Union[BaseBackend, Backend]]:
+        """Return backend.
+
+        Returns:
+            Backend this experiment is for, or ``None`` if backend is unknown.
+        """
+        return self._backend
+
+    @property
+    def experiment_type(self) -> str:
+        """Return experiment type.
+
+        Returns:
+            Experiment type.
+        """
+        return self._type
+
+    @property
+    def figure_names(self) -> List[str]:
+        """Return names of the figures associated with this experiment.
+
+        Returns:
+            Names of figures associated with this experiment.
+        """
+        return self._figures.keys()
+
+    @property
+    def share_level(self) -> str:
+        """Return the share level fo this experiment.
+
+        Returns:
+            Experiment share level.
+        """
+        return self._share_level
+
+    @share_level.setter
+    def share_level(self, new_level: str) -> None:
+        """Set the experiment share level.
+
+        Args:
+            new_level: New experiment share level. Valid share levels are provider-
+                specified. For example, IBM Quantum experiment service allows
+                "global", "hub", "group", "project", and "private".
+        """
+        self._share_level = new_level
+        if self.auto_save:
+            self.save()
+
+    @property
+    def notes(self) -> str:
+        """Return experiment notes.
+
+        Returns:
+            Experiment notes.
+        """
+        return self._notes
+
+    @notes.setter
+    def notes(self, new_notes: str) -> None:
+        """Update experiment notes.
+
+        Args:
+            new_notes: New experiment notes.
+        """
+        self._notes = new_notes
+        if self.auto_save:
+            self.save()
+
+    @property
+    def service(self) -> Optional[DatabaseServiceV1]:
+        """Return the database service.
+
+        Returns:
+            Service that can be used to access this experiment in a database.
+        """
+        return self._service
+
+    @service.setter
+    def service(self, service: DatabaseServiceV1) -> None:
+        """Set the service to be used for storing experiment data.
+
+        Args:
+            service: Service to be used.
+
+        Raises:
+            DbExperimentDataError: If an experiment service is already being used.
+        """
+        if self._service:
+            raise DbExperimentDataError("An experiment service is already being used.")
+        self._service = service
+        with contextlib.suppress(Exception):
+            self.auto_save = self._service.options.get("auto_save", False)
+
+    @property
+    def source(self) -> Dict:
+        """Return the class name and version."""
+        return self._source
+
+    def __str__(self):
```

Comment:
There are some fields that are missing in here it seems, like share level, backend, tags, notes - are those intentionally omitted?

Reply:
It was intentional since I didn't want the output to be a page long lol. Having said that, `backend` and `tags` seem important enough to be included, so I added them in 2e01e94.